apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -1,46 +1,587 @@
+	Inspect     OperationMetric `json:"inspect,omitempty"`
+	Provision   OperationMetric `json:"provision,omitempty"`
+	Deprovision OperationMetric `json:"deprovision,omitempty"`
 }
 
 // BareMetalHostStatus defines the observed state of BareMetalHost
 type BareMetalHostStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Important: Run "operator-sdk generate k8s" to regenerate code
+	// after modifying this file
+
+	// OperationalStatus holds the status of the host
+	// +kubebuilder:validation:Enum="";OK;discovered;error
+	OperationalStatus OperationalStatus `json:"operationalStatus"`
+
+	// ErrorType indicates the type of failure encountered when the
+	// OperationalStatus is OperationalStatusError
+	// +kubebuilder:validation:Enum=registration error;inspection error;provisioning error;power management error
+	ErrorType ErrorType `json:"errorType,omitempty"`
+
+	// LastUpdated identifies when this status was last observed.
+	// +optional
+	LastUpdated *metav1.Time `json:"lastUpdated,omitempty"`
+
+	// The name of the profile matching the hardware details.
+	HardwareProfile string `json:"hardwareProfile"`
+
+	// The hardware discovered to exist on the host.
+	HardwareDetails *HardwareDetails `json:"hardware,omitempty"`
+
+	// Information tracked by the provisioner.
+	Provisioning ProvisionStatus `json:"provisioning"`
+
+	// the last credentials we were able to validate as working
+	GoodCredentials CredentialsStatus `json:"goodCredentials,omitempty"`
+
+	// the last credentials we sent to the provisioning backend
+	TriedCredentials CredentialsStatus `json:"triedCredentials,omitempty"`
+
+	// the last error message reported by the provisioning subsystem
+	ErrorMessage string `json:"errorMessage"`
+
+	// indicator for whether or not the host is powered on
+	PoweredOn bool `json:"poweredOn"`
+
+	// OperationHistory holds information about operations performed
+	// on this host.
+	OperationHistory OperationHistory `json:"operationHistory"`
+
+	// ErrorCount records how many times the host has encoutered an error since the last successful operation
+	ErrorCount int `json:"errorCount"`
 }
 
-// +kubebuilder:object:root=true
+// ProvisionStatus holds the state information for a single target.
+type ProvisionStatus struct {
+	// An indiciator for what the provisioner is doing with the host.
+	State ProvisioningState `json:"state"`
+
+	// The machine's UUID from the underlying provisioning tool
+	ID string `json:"ID"`
+
+	// Image holds the details of the last image successfully
+	// provisioned to the host.
+	Image Image `json:"image,omitempty"`
+
+	// The RootDevicehints set by the user
+	RootDeviceHints *RootDeviceHints `json:"rootDeviceHints,omitempty"`
+
+	// BootMode indicates the boot mode used to provision the node
+	BootMode BootMode `json:"bootMode,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
Perhaps? I just moved the old file into the new location, and I probably should have done what I did with the controller where I copied the contents over by hand in pieces.

Reply:
Further research reveals that they're probably not related, but the deepcopy-gen thing was to cause it to generate a DeepCopyObject() method, which was some [ancient](https://www.openshift.com/blog/kubernetes-deep-dive-code-generation-customresources) workaround that doesn't appear to be needed any more since [we are generating this anyway](https://github.com/metal3-io/baremetal-operator/pull/655/files#diff-682ec4bfea44947253d1b0be36f05931R78-R83).